src/Nethermind/Nethermind.Consensus.AuRa/Contracts/RewardContract.cs

```diff
@@ -117,11 +111,11 @@ public RewardContract(IAbiEncoder abiEncoder, Address contractAddress, long tran
         /// 101-106 - Uncle - Reward attributed to uncles, with distance 1 to 6 (Ethash engine)
         /// </param>
         public Transaction Reward(Address[] benefactors, ushort[] kind)
-            => GenerateTransaction(_abiEncoder.Encode(Definition.reward, benefactors, kind), Address.SystemUser);
+            => GenerateSystemTransaction(_abiEncoder.Encode(Definition.Functions[RewardFunction].GetCallInfo(), benefactors, kind));
```

Comment:
have you thouhg of DLR and Expando objects here?

Reply:
Yes it was dictionary vs casting, went with dictionary